File: common/src/main/java/com/demo/common/dto/UserDto.java
package com.demo.common.dto;

import lombok.*;

import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserDto {
    private UUID id;
    private String name;
    private String email;
    private String username;
    private boolean active;
    /** UUID of the file-service record for the user's profile picture; null if not set. */
    private UUID avatarFileId;

    /** ISO 639-1 language code for the user's preferred UI language (e.g. "en", "hr"). */
    private String language;

    /** Manageable Keycloak realm roles held by this user (excludes WEB_APP). Empty for list endpoints. */
    private List<String> roles;
}

File: android/data/src/main/kotlin/com/demo/taskmanager/data/dto/UserDto.kt
package com.demo.taskmanager.data.dto

import kotlinx.serialization.Serializable

@Serializable
data class UserDto(
    val id: String,
    val name: String,
    val email: String,
    val username: String,
    val active: Boolean,
    /** File-service UUID for the user's profile picture; null when not set. */
    val avatarFileId: String?,
    /** ISO 639-1 language code (e.g. "en", "hr"). */
    val language: String?,
    /** Keycloak realm roles held by this user; empty on list endpoints. */
    val roles: List<String> = emptyList(),
)

@Serializable
data class UserRoleDto(
    val roles: List<String>,
)

@Serializable
data class UserCreateRequest(
    val name: String,
    val email: String,
    val username: String,
    val active: Boolean = true,
)

@Serializable
data class UserUpdateRequest(
    val name: String? = null,
    val email: String? = null,
    val username: String? = null,
    val active: Boolean? = null,
)
